### DISCUSSION TITLE
Unable to promote a project to production

### DISCUSSION THREAD

**Discussion by pflooky** [vercel/community#5684] [Help] [answered]
### Summary

Unable to promote a project to production. I get the error "An unexpected internal error occurred" once I try to click on "Promote". I assume it is due to me deleting a previous deployment that was deployed to production.

### Steps to Reproduce

![Screen Shot 2024-01-31 at 12 53 43 PM](https://github.com/vercel/community/assets/26299147/ea116e4d-fc7e-443d-9c9a-4b1e2a00da9a)
![Screen Shot 2024-01-31 at 12 53 57 PM](https://github.com/vercel/community/assets/26299147/bd69559a-d5dc-41db-9e51-d7d2aebf74d7)
![Screen Shot 2024-01-31 at 12 54 13 PM](https://github.com/vercel/community/assets/26299147/40d58dff-625d-41fc-a164-1387db360e1f)

**Comment by jm42** [comment]
I am having the same problem. All system operational but still this happens. Probably a bug.

**Comment by swarnava** [comment]
Can you share a screencast which we can look into? Also are you seeing any failed request in Network log?

**Reply by jm42** [reply]
Indeed. An error 500 to `POST https://vercel.com/api/v9/projects/prj_foo/promote/dpl_bar?teamId=team_baz` that returns:

```json
{
	"error": {
		"code": "internal_server_error",
		"message": "An unexpected internal error occurred"
	}
}
```

**Reply by swarnava** [reply]
are you able to `promote` using CLI command? https://vercel.com/docs/cli/promote

**Reply by jm42** [reply]
No, it doesn't. Same error 500 is thrown:

```
> [debug] [2024-03-21T11:49:58.599Z] Retrying: Error: An unexpected internal error occurred (500)
Error: An unexpected internal error occurred (500)
    at responseError (/home/jm/.config/yarn/global/node_modules/vercel/dist/index.js:99959:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/jm/.config/yarn/global/node_modules/vercel/dist/index.js:101234:28
    at async requestPromote (/home/jm/.config/yarn/global/node_modules/vercel/dist/index.js:195658:3)
    at async promote_default (/home/jm/.config/yarn/global/node_modules/vercel/dist/index.js:195809:16)
    at async main13 (/home/jm/.config/yarn/global/node_modules/vercel/dist/index.js:199272:18)
> [debug] [2024-03-21T11:50:02.603Z] #8 → POST https://api.vercel.com/v9/projects/prj_foo/promote/dpl_bar?teamId=team_baz
> [debug] [2024-03-21T11:50:02.858Z] #8 ← 500 Internal Server Error: sfo1::v4xvz-1711021802690-eef91e3d1ca0 [255ms]
```

**Reply by swarnava** [reply]
Can you check if you have `production` domain added in your project?

**Reply by jm42** [reply, marked as answer]
I fix it by removing the production domain and creating it again. Something was odd with the domain.
